osu.Game/Overlays/ControllableOverlayHeader.cs

```diff
@@ -0,0 +1,46 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Framework.Allocation;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Shapes;
+using osu.Framework.Graphics.UserInterface;
+using osu.Game.Graphics;
+using osuTK.Graphics;
+
+namespace osu.Game.Overlays
+{
+    /// <typeparam name="T">The type of item to be represented by tabs in <see cref="TabControl{T}"/>.</typeparam>
+    public abstract class ControllableOverlayHeader<T> : OverlayHeader
```

Comment:
please add full xmldoc. also reconsider the naming – this contains a `TabControl` so why is this class not called `TabControlOverlayHeader` rather than an implementation?

Reply:
Well, `TabControlOverlayHeader` is already used. And honestly I have no idea how to rename it. If we will use `TabControlOverlayHeader` for this one - how should I name the original one?

Reply:
Nevermind, I figured that out